connectors.md

```diff
@@ -1,26 +1,90 @@
 # Connectors and models
 
-## What is a connector?
+This document is intended as a design document for people who want to write connectors for various backends. Its main purpose is to specify what properties connectors should have so they can be easily shared with other people and used in Apollo without any shims.
```

Comment:
I think the beginning could have a small bullet list like:

- **Connector**: a database driver that has GraphQL-specific error handling, logging, batching, and/or caching. These can be shipped as general packages for each data source.
- **Model**: a set of functions to fetch data of a certain type that calls into the connectors. These are almost always application-specific.